sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -164,4 +172,78 @@ private object PostgresDialect extends JdbcDialect {
     s"TABLESAMPLE BERNOULLI" +
       s" (${(sample.upperBound - sample.lowerBound) * 100}) REPEATABLE (${sample.seed})"
   }
+
+  // CREATE INDEX syntax
+  // https://www.postgresql.org/docs/14/sql-createindex.html
+  override def createIndex(
+      indexName: String,
+      tableName: String,
+      columns: Array[NamedReference],
+      columnsProperties: util.Map[NamedReference, util.Map[String, String]],
+      properties: util.Map[String, String]): String = {
```

Comment:
The implementation is similar to `MySQL`.  We probably want to move this to `JdbcUtils` too for code reuse.

Reply:
@huaxingao The implementation is not totally similar to `MySQL` in the assignment of variables. So we should create a function with the type of JDBC parameter that we base on it to return the exact SQL, right? Thanks!